src/construct_realm.js

```diff
@@ -17,9 +17,19 @@ import * as evaluators from "./evaluators/index.js";
 import * as partialEvaluators from "./partial-evaluators/index.js";
 import { NewGlobalEnvironment } from "./methods/index.js";
 import { ObjectValue } from "./values/index.js";
+import { Debugger } from "./debugger/Debugger.js";
+import fs from "fs";
 
 export default function(opts: RealmOptions = {}): Realm {
   let r = new Realm(opts);
+  if (r.enableDebugger) {
+    let dbgFileContents = fs.readFileSync("./src/debugger/.sessionlogs/proxy2debugger.txt", "utf8");
+    let dbgFileLines = dbgFileContents.toString().split("\n");
+    if (dbgFileLines[0] === "Debugger Attached") {
```

Comment:
High level, creating Debugger instance should be associated with "enableDebugger" flag, while "Debugger Attached" should be treated as a normal debugger command and processed in Debugger.js.

Reply:
Why split dbgFileContents? Also, under what circumstances would we supply the realm with a debug channel to which no debugger is attached?